syft/pkg/cataloger/java/parse_pom_xml.go

```diff
@@ -109,3 +113,11 @@ func cleanDescription(original string) (cleaned string) {
 	}
 	return strings.TrimSpace(cleaned)
 }
+
+func propertyReplacer(pomProperties gopom.Properties, stringWithVariablesToReplace string) string {
+	propertyMatcher := regexp.MustCompile("[$][{][^}]+[}]")
+	return propertyMatcher.ReplaceAllStringFunc(stringWithVariablesToReplace, func(match string) string {
+		prop := match[2 : len(match)-1]
+		return pomProperties.Entries[prop]
```

Comment:
Thinking about this a little more, it might be good to only return this value if it's found in the map... so we can see which properties are not found more obviously. This would be something like:
```go
value, ok := pomProperties.Entries[prop]
if ok {
  return value
}
return match
```
The main reason I'd prefer to do this is because variables might be coming from a parent pom or another expression like `${project.groupId}`, which we're not currently processing and if these values end up blank, it is less obvious that there's an expression we might need to support.